Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm

```diff
@@ -129,6 +129,46 @@
 "</script>"
 "</html>";
 
+static constexpr Seconds actionAbsenceTimeout = 300_ms;
+static constexpr Seconds actionDoneTimeout = 1000_ms;
+
+namespace {
+
+class ScopedNotificationCenterObserveOnce {
+public:
+    ScopedNotificationCenterObserveOnce() = default;
+    ScopedNotificationCenterObserveOnce(NSNotificationName name, NSObject* object)
+    {
+        m_observer = [m_center addObserverForName:name object:object queue:nil usingBlock:[this] (NSNotification *) {
+            [m_center removeObserver:m_observer.get()];
+            m_observer = nullptr;
+            m_didObserve = true;
+            m_center = nullptr;
+        }];
+    }
+    ~ScopedNotificationCenterObserveOnce()
+    {
+        if (!m_observer)
+            return;
+        [m_center removeObserver:m_observer.get()];
+    }
+    ScopedNotificationCenterObserveOnce(ScopedNotificationCenterObserveOnce&&) = default;
+    ScopedNotificationCenterObserveOnce& operator=(ScopedNotificationCenterObserveOnce&&) = default;
```

Comment:
Why did you have to write these? Won’t it compile and do exactly the same thing without explicitly defining them?